src/CodeGenerator/src/Generator/RequestSerializer/RestJsonSerializer.php

```diff
@@ -171,18 +171,32 @@ private function dumpArrayList(string $output, string $input, string $contextPro
         $memberShape = $shape->getMember()->getShape();
 
         return strtr('
-            $index = -1;
-            foreach (INPUT as $listValue) {
-                $index++;
-                MEMBER_CODE
+            if (null !== INPUT) {
+                $index = -1;
+                INIT_CODE
+                foreach (INPUT as $listValue) {
+                    $index++;
+                    MEMBER_CODE
+                }
             }
         ',
             [
                 'INPUT' => $input,
+                'INIT_CODE' => $this->dumpArrayInit($output, $memberShape),
                 'MEMBER_CODE' => $memberCode = $this->dumpArrayElement(sprintf('%s[$index]', $output), '$listValue', $contextProperty, $memberShape, true),
             ]);
     }
 
+    private function dumpArrayInit(string $output, Shape $shape)
+    {
+        switch (true) {
+            case $shape instanceof StructureShape && 'AttributeValue' === $shape->getName():
```

Comment:
remove specific code `AttributeValue`

Reply:
Same here. I restricted to reduce side effect and broken test.

I will revert it to let you have a look on broken test?